app/javascript/lib/search-metrics.js

```diff
@@ -0,0 +1,275 @@
+    // these artifactual searches shifts that cost-benefit, somewhat.
+    return
+  }
+
+  numSearches += 1 // Number of searches insofar as usage analytics is concerned
+
+  const {
+    terms, termString, numTerms,
+    genes, geneString, numGenes,
+    numFacets, numFilters, facetList, filterListByFacet
+  } = getSearchQueryLogProps(searchParams.terms, searchParams.genes, searchParams.facets)
+
+  const page = searchParams.page
+  const preset = searchParams.preset
+  const scpStudiesMatchData = searchResults?.matchByData
+
+  const simpleProps = {
+    terms, termString, numTerms, genes, geneString, numGenes, page, preset,
+    facetList, numFacets, numFilters,
+    perfTimes, numSearches,
+    type, context: 'global'
+  }
+  const props = Object.assign(simpleProps, filterListByFacet, scpStudiesMatchData)
+
+  log('search', props)
+
+  let gaEventCategory = 'advanced-search'
+  // e.g. advanced-search-covid19
+  if (
+    preset !== '' &&
+    typeof preset !== 'undefined'
+  ) {
+    gaEventCategory += `-${preset}`
+  }
+
+  // Google Analytics fallback: remove once Bard and Mixpanel are ready for SCP
+  ga( // eslint-disable-line no-undef
+    'send', 'event', gaEventCategory, 'study-search',
+    'num-terms', numTerms
+  )
+}
+
+/**
+ * Log analytics about search result selection to Mixpanel
+ *
+ * This logging differentiates properties about the selected result from
+ * properties about other aspects of the search -- e.g. the full result set
+ * and query -- by prefixing the latter with a "namespace", i.e. `results:`.
+ *
+ * @param {Object} study Selected result
+ * @param {Object} logProps Other properties to log
+ *
+ * TODO (SCP-4256): Refine search result selection logging
+ * TODO (SCP-4257): Eliminate duplicate default/custom search event logging
+ */
+export function logSelectSearchResult(study, logProps={}) {
+  logProps = Object.assign(
+    {
+      studyAccession: study.accession,
+      context: 'study',
+      scope: 'global'
+    },
+    study,
+    logProps
+  )
+  delete logProps.accession // Redundant with more-conventional studyAccession
+
+  Object.entries(logProps.results).forEach(([key, value]) => {
+    if (['termList', 'geneList'].includes(key)) {
+      // Redundant with pre-existing `terms` and `genes` Mixpanel props
+      return
+    }
+    logProps[`results:${key}`] = value
+  })
+  delete logProps.results // Remove nested property; flattened above
+
+  // Number of searches done before this result was selected.
+  // This is needed to easily analyze changes in click-through rate (CTR).
+  logProps.numSearches = numSearches
+
+  // Number of selections done for the current search.  Users can click
+  // multiple results for a given search, e.g. via command key + click, and
+  // this property lets us account for that when analyzing CTR and related
+  // metrics.
+  if (numSearches !== prevNumSearches) {
+    numSearchSelections = 1
+    prevNumSearches = numSearches
+  }
```

Comment:
```suggestion
  if (numSearches !== prevNumSearches) {
    numSearchSelections = 1
    prevNumSearches = numSearches
  }
```
I think you can delete this if block if numSearchSelections is reset inside `logSearch`